Added new resources & updated existing ones

feyzee opened the pull request:
I've added resources for IAM policies that are attached with Lambda's role. I've also added terraform cloudwatch logs resource and test data as json files for Lambda functions.
Lambda functions have updated to include exception handling for boto3 sdk calls. And moved ndjson loop to a new function.

VikrantSinghBhadauriya reviewed (commented):
IAM policy for lamda will also incluse access to S3?

feyzee commented:
Only the second Lambda requires access to S3 for GET and PUT operations. In the first Lambda function all the required information is passed as event data.

I'll update the code to include that format. There is a `aws_cloudwatch_log_resource_policy` resource in terraform which I avoided using. It could allow creation of the log group without needing to explicitly mentioning in IAM policy, but there seems to be some sort of hard limit(upto 10 requests) for that account wide.

feyzee commented:
Updated the name of cloudwatch resource. Since we cannot access function name using the `aws_lambda_function` resource attribute, I've created a variable with function name as value and updated the function name in `aws_lambda_function` resource.
